### PR TITLE
Expose base.py as a public api.

### PR DESCRIPTION
**Pull request by arenevier**
Extract methods that can be used to implement Websocket over HTTP/2

The implementation of HTTP/2 handshake is quite similar to HTTP/1 with a
few differences:
- headers are lower case in HTTP/2
- not exactly the same headers must be validated
- We don't handle Sec-WebSocket-Accept for HTTP/2
- the response is different

So we can implement the handshake in a base class that will delegate
some operations to the parent class.

We make _base.py public by renaming it to base.py

### CODE REVIEW

**Comment by arenevier** [comment]
two notes:
- I decide to rename _base.py to base.py because: a) it now contains a base class b) there is already an util.py in mod_pywebsocket/ so I figured it would be confusing
- Here is what HTTP/2 handshaker would look like in wpt with that change: https://gist.github.com/arenevier/1845daecc3dccad278b17e7015f5aa8f#file-ws_h2_handshake-py

File: mod_pywebsocket/handshake/base.py [hunk 1]
```diff
@@ -0,0 +1,411 @@
+# Copyright 2012-2021, Google Inc.
```
Comment (ricea) [inline mod_pywebsocket/handshake/base.py:1]: No need to update the copyright date (unless you want to).

File: mod_pywebsocket/handshake/base.py [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+    fields = request.headers_in[common.HOST_HEADER].split(':', 1)
+    if len(fields) == 1:
+        return fields[0], get_default_port(request.is_https())
+    try:
+        return fields[0], int(fields[1])
+    except ValueError as e:
+        raise HandshakeException('Invalid port number format: %r' % e)
+
+
+def format_header(name, value):
```
Comment (ricea) [inline mod_pywebsocket/handshake/base.py:129]: We can move this to hybi.py, as it's only used there.

File: mod_pywebsocket/handshake/base.py [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+    value = get_mandatory_header(request, key)
+
+    if value.lower() != expected_value.lower():
+        raise HandshakeException(
+            'Expected %r for header %s but found %r (case-insensitive)' %
+            (expected_value, key, value),
+            status=fail_status)
+
+
+def check_request_line(request):
```
Comment (ricea) [inline mod_pywebsocket/handshake/base.py:150]: We can also move this to hybi.py.

**Review by ricea — COMMENTED** [review]
lgtm with tiny nits.

File: mod_pywebsocket/handshake/base.py (hunk 2 above)
Comment (arenevier) [reply mod_pywebsocket/handshake/base.py]: done

File: mod_pywebsocket/handshake/base.py (hunk 3 above)
Comment (arenevier) [reply mod_pywebsocket/handshake/base.py]: done

**Review by ricea — APPROVED** [review]
lgtm

**Pull request merged**